src/services/item/plugins/published/service.ts

```diff
@@ -123,7 +128,16 @@ export class ItemPublishedService {
     const item = await this.itemService.get(actor, repositories, itemId, PermissionLevel.Admin);
 
     // item should be public first
-    await itemTagRepository.getType(item, ItemTagType.Public, { shouldThrow: true });
+    const tag = await itemTagRepository.getType(item, ItemTagType.Public, {
+      shouldThrow: shouldBePublic,
+    });
+
+    // if the item can be private and be published, set it to public automatically.
+    // it's usefull for the publishing automatically after the validation.
+    // the user is asked to set the item to public in the frontend.
+    if (!tag && !shouldBePublic) {
+      await itemTagRepository.post(actor, item, ItemTagType.Public);
```

Comment:
It is the responsibility of the publish service or the validation service to set the item to public if it is private ?